src/meta/typesmap.go

```diff
@@ -311,6 +311,11 @@ func (m *TypesMap) IsEmpty() bool {
 	return len(m.m) == 0
 }
 
+// Equals check if two typesmaps are the same
+func (m *TypesMap) Equals(m2 *TypesMap) bool {
```

Comment:
This is a very inefficient way to compare types.

Reply:
Should be fixed.

Reply:
m can also be nil. There are no checks for this in String() method probably just because I didn't use it anywhere (it is there for debugging purposes but there is comment about that sadly).

Reply:
You can iterate nil maps and get their length (it will be 0). So you only need to check that m or m2 are nil. Two empty type sets can be equal as well so it's even incorrect to return false when they are nil. Also m.m will still cause panic if m is nil.

Reply:
Updated. As of #257 we are not using pointers here which simplifies things a little.